src/commons/eventsourcing/wal.rs

```diff
@@ -120,6 +120,7 @@ impl<T: WalSupport> WalSet<T> {
 pub struct WalStore<T: WalSupport> {
     kv: KeyValueStore,
     cache: RwLock<HashMap<MyHandle, Arc<T>>>,
+    locks: HandleLocks,
```

Comment:
Commenting here since https://github.com/github/roadmap/issues/347 is still open.
Is the purpose of the lock to protect access to `kv`? If so, shouldn't the lock be acquired in more functions that use `kv`, I see some functions do not acquire the lock, e.g. `has()`,  `get_snapshot()` and `list()`?

Reply:
No, the KV store is used for different entities identified by their own handles. So, we specifically want to lock operations on a specific handle only. E.g. otherwise CA1 would have to wait for CA2's ROA changes to complete etc. This is even worse when we have an embedded parent CA that can be busy.

Operations on different handles can never conflict, so I did not want the global lock.